components/edit_channel_header_modal/edit_channel_header_modal.jsx

```diff
@@ -106,7 +107,7 @@ class EditChannelHeaderModal extends React.PureComponent {
     }
 
     focusTextbox = () => {
-        if (!Utils.isMobile()) {
+        if (this.refs.editChannelHeaderTextbox) {
```

Comment:
This is the fix. I can't find a reason why previously we prevented focus whenever it's on mobile.  Now, this has similar behavior with `EditChannelPurposeModal`.